### PR TITLE
#882 Editing WP Status should no longer be allowed

### LINKED ISSUE #882: Work Packages - Should not be able to edit status
### Description

Work package status should not be editable. It should onyl be changed by activation adn stage gate change requests

### Acceptance Criteria

It should be removed from the entire work package editing flow

### Proposed Solution

- [ ] remove it from the edit form
- [ ] remove it from the payload sent to the backend
- [ ] remove it from the route validation on the backend
- [ ] remove it from the controller function
- [ ] remove it from the service function
- [ ] remove it from tests if needed (aka anywhere the service function is called)

### Mocks

_No response_

### PR DESCRIPTION
**Pull request by nezamj03**
## Changes

Removed Status from form, from payload, and from route, controller, and service.

## Screenshots

Screen Shot 2023-02-27 at 2 51 24 AM

## To Do

_Any remaining things that need to get done_

- [ ] write tests for edit WP

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please review the [contributor guide](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/8060929/Software+Contributor+Guide) and reach out to your Tech Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Screenshots of UI changes (see Screenshots section)
- [x] Remove any non-applicable sections of this template
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #882

### CODE REVIEW

File: src/backend/src/services/work-packages.services.ts [hunk 1]
```diff
@@ -34,7 +33,6 @@ export default class WorkPackagesService {
    * @returns a list of work packages
    */
   static async getAllWorkPackages(query: {
-    status?: WbsElementStatus;
```
Comment (anthonybernardi) [inline src/backend/src/services/work-packages.services.ts:37]: this stuff is definitely still needed, this is how we get work packages of a certain status

File: src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditDetails.tsx [hunk 2]
```diff
@@ -79,9 +79,6 @@ const WorkPackageEditDetails: React.FC<Props> = ({ users, control, errors }) =>
             )}
           />
         </Grid>
-        <Grid item xs={12} md={4} sx={{ mt: 2, mb: 1 }}>
-          <StatusSelect />
```
Comment (anthonybernardi) [inline src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditDetails.tsx:83]: need to delete StatusSelect the component too

**Review by anthonybernardi — CHANGES_REQUESTED** [review]
this doesn't compile for me because you need to update the calls about status in the seed data and possible other places

File: src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditDetails.tsx [hunk 3]
```diff
@@ -96,52 +72,37 @@ const WorkPackageEditDetails: React.FC<Props> = ({ users, control, errors }) =>
...
-                      ))}
-                    </TextField>
-                  </>
-                )}
-              />
-            </FormControl>
-          </Grid>
-          <Grid item xs={6}>
-            <FormControl sx={{ width: '90%' }}>
-              <FormLabel>Project Manager</FormLabel>
```
Comment (RChandler234) [inline src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditDetails.tsx:126]: add the form labels back in for project lead and manager

**Review by RChandler234 — CHANGES_REQUESTED** [review]
image
make sure the spacing looks good on these (make sure there's a little gap between project lead and project manager, and you can probably fit stage select where status select used to be)

**Review by anthonybernardi — APPROVED** [review]
awesome!!

**Pull request merged**